cms/djangoapps/contentstore/views/item.py

```diff
@@ -145,12 +146,25 @@ def xblock_handler(request, usage_key_string):
             accept_header = request.META.get('HTTP_ACCEPT', 'application/json')
 
             if 'application/json' in accept_header:
+                store = modulestore()
                 fields = request.GET.get('fields', '').split(',')
+                format = request.GET.get('format', '')
                 if 'graderType' in fields:
                     # right now can't combine output of this w/ output of _get_module_info, but worthy goal
                     return JsonResponse(CourseGradingModel.get_section_grader_type(usage_key))
+                elif format == 'concise':
+                    xblock = _get_xblock(usage_key, request.user)
+                    course = store.get_course(xblock.location.course_key)   # pylint: disable=no-member
+                    course_outline = create_xblock_info(
+                        course,
+                        include_child_info=True,
+                        include_children_predicate=lambda xblock: xblock.has_children,
+                        is_concise=format
+                    )
+                    ancestor_info = _create_xblock_ancestor_info(xblock, course_outline, is_concise=format)
```

Comment:
> Another option would be to issue two requests: one to get the course outline, and a second one to get the current location. In some ways this is better because the two pieces of information are used by different parts of the UI, so it improves the modularity.

This looks a good option.

But the thing is `ancestor_info = _create_xblock_ancestor_info(xblock, course_outline, is_concise=format)` requires `course_outline`. So would we be sending the course_outline back from JS ? I don't think we should do that.

Also, another option that I looked for is that we calculate this in front-end but that would be checking entire course_outline each time a new listing of the tree nodes is shown, it may be a bit expensive when compared to one-time backend call ?